### PR TITLE
Deploy archives without publication in CI mode

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](../CONTRIBUTING.md) passed. If this feature is not already covered by the tests, I added new
  tests.

-----

CI mode - deploy archives when no publishing extension is applied.
This PR is necessary to upgrade the Gradle plugin to 5+ in the JFrog CLI and in the Jenkins Artifactory plugin.

### CODE REVIEW

File: src/main/java/org/jfrog/gradle/plugin/artifactory/utils/PublicationUtils.java [hunk 1]
```diff
@@ -255,6 +260,42 @@ public static void extractMavenDeployDetails(MavenPublicationInternal mavenPubli
...
+            }
+            String gid = project.getGroup().toString();
+            if (publisher.isM2Compatible()) {
+                gid = gid.replace(".", "/");
+            }
+            Map<String, String> extraTokens = new HashMap<>();
+            if (StringUtils.isNotBlank(artifact.getClassifier())) {
+                extraTokens.put("classifier", artifact.getClassifier());
+            }
+            String artifactPath = IvyPatternHelper.substitute(publisher.getIvyArtifactPattern(), gid, project.getName(),
```
Comment (attiasas) [inline src/main/java/org/jfrog/gradle/plugin/artifactory/utils/PublicationUtils.java:287]: IvyArtifactPattern can be optional right? Should we check for that before?

**Review by attiasas — CHANGES_REQUESTED** [review]
looks good! consider the following:
1. adding tests that cover those cases.
2. is this option only available in CI mode? can't we extend it for normal use as well? maybe with a flag if needed
3. update README configuration part to include this as a note

File: src/main/java/org/jfrog/gradle/plugin/artifactory/utils/PublicationUtils.java (hunk 1 above)
Comment (yahavi) [reply src/main/java/org/jfrog/gradle/plugin/artifactory/utils/PublicationUtils.java:287]: getIvyArtifactPattern` returns a default non-blank value if the IvyArtifactPattern is blank:
```java
public String getIvyArtifactPattern() {
    String value = getStringValue(IVY_ART_PATTERN);
    if (StringUtils.isBlank(value)) {
        return LayoutPatterns.M2_PATTERN;
    }
    return value.trim();
}
```

**Comment by yahavi** [comment]
1. Added ciServerArchivesTest
2. This shouldn't work in a non-ci mode. We should encourage users to use only Gradle publications.
3. But there is zero configuration needed. The process is entirely transparent for the user. Also, users of this plugin will not execute the added code.

**Pull request merged**